website/source/docs/platform/k8s/service-sync.html.md

```diff
@@ -0,0 +1,274 @@
+the Kubernetes cluster is generally easier since it is automated using the
+[Helm chart](/docs/platform/k8s/helm.html).
+
+To install the sync, enable the catalog sync feature using
+[Helm values](/docs/platform/k8s/helm.html#configuration-values-) and
+upgrade the installation using `helm upgrade` for existing installs or
+`helm install` for a fresh install.
+
+```yaml
+syncCatalog:
+  enabled: true
+```
+
+This will enable services to sync _in both directions_. You can also choose
+to only sync Kubernetes services to Consul or vice versa by disabling a direction.
+See the [Helm configuration](/docs/platform/k8s/helm.html#configuration-values-)
+for more information.
+
+-> **Before installing,** please read the introduction paragraphs for the
+reference documentation below for both
+[Kubernetes to Consul](/docs/platform/k8s/service-sync.html#kubernetes-to-consul) and
+[Consul to Kubernetes](/docs/platform/k8s/service-sync.html#consul-to-kubernetes)
+sync to understand how the syncing works.
+
+### Authentication
+
+The sync process must authenticate to both Kubernetes and Consul to read
+and write services.
+
+For Consul, the process accepts both the standard CLI flag `-token` and
+the environment variable `CONSUL_HTTP_TOKEN`. This should be set to an
+Consul [ACL token](/docs/guides/acl.html) if ACLs are enabled. This
+can also be configured using the Helm chart to read from a Kubernetes
+secret.
+
+For Kubernetes, a valid kubeconfig file must be provided with cluster
+and auth information. The sync process will look into the default locations
+for both in-cluster and out-of-cluster authentication. If `kubectl` works,
+then the sync program should work.
+
+## Kubernetes to Consul
+
+This sync registers Kubernetes services to the Consul catalog automatically.
+
+This enables discovery and connection to Kubernetes services using native
+Consul service discovery such as DNS or HTTP. This is particularly useful for
+non-Kubernetes nodes. This also causes all discoverable services to be part of
+a central service catalog in Consul for further syncing into alternate
+Kubernetes clusters or other platforms.
+
+### Kubernetes Service Types
+
+Not all Kubernetes services are externally accessible. The sync program by
+default will only sync services of the following types or configurations.
+If a service type is not listed below, then the sync program will ignore that
+service type.
+
+#### NodePort
+
+[NodePort services](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport)
+register a static port that every node in the K8S cluster listens on.
+
+For NodePort services, a Consul service instance will be created for each
+node that has the representative pod running. While Kubernetes configures
+a static port on all nodes in the cluster, this limits the number of service
+instances to be equal to the nodes running the target pods.
+
+The service instances registered will be registered to the K8S node name
```

Comment:
Is there an extra `registered` in this sentence? Reading it, it makes more sense as "The service instances will be registered to the K8s node name..."